rpcs3/Emu/Io/GameTablet.cpp

```diff
@@ -0,0 +1,281 @@
+	// Interrupt transfers are slow (6ms, TODO accurate measurement)
+	transfer->expected_time = get_timestamp() + 6000;
+
+	memset(buf, 0, buf_size);
+
+	buf[0x02] = 0x0f; // dpad
+
+	buf[0x03] = 0x80;
+	buf[0x04] = 0x80;
+	buf[0x05] = 0x80;
+	buf[0x06] = 0x80;
+
+	buf[0x0d] = 0x72; // pressure
+	buf[0x0f] = 0x0f; // pos X
+	buf[0x10] = 0x0f; // pos Y
+	buf[0x11] = 0xff;
+	buf[0x12] = 0xff;
+
+	buf[0x13] = 0x01; // accel X
+	buf[0x14] = 0x02;
+	buf[0x15] = 0x00; // accel Y
+	buf[0x16] = 0x02;
+	buf[0x17] = 0xea; // accel Z
+	buf[0x18] = 0x01;
+
+	buf[0x1a] = 0x02;
+
+	if (!is_input_allowed())
+	{
+		return;
+	}
+	if (g_cfg.io.mouse != mouse_handler::basic)
+	{
+		gametablet_log.warning("GameTablet requires mouse_handler configured to basic");
+		return;
+	}
+
+	std::lock_guard lock(pad::g_pad_mutex);
+	const auto gamepad_handler = pad::get_current_handler();
+	const auto& pads = gamepad_handler->GetPads();
+
+	bool up, right, down, left;
+	up = right = down = left = false;
+
+	const int padIndex = 1; // Player2
+	const auto& pad = pads[padIndex];
+	if (pad->m_port_status & CELL_PAD_STATUS_CONNECTED)
+	{
+		for (Button& button : pad->m_buttons)
+		{
+			if (!button.m_pressed)
+			{
+				continue;
+			}
+
+			if (button.m_offset == CELL_PAD_BTN_OFFSET_DIGITAL1)
+			{
+				switch (button.m_outKeyCode)
+				{
+				case CELL_PAD_CTRL_SELECT:
+					buf[1] |= (1 << 0);
+					break;
+				case CELL_PAD_CTRL_START:
+					buf[1] |= (1 << 1);
+					break;
+				case CELL_PAD_CTRL_UP:
+					up = true;
+					break;
+				case CELL_PAD_CTRL_RIGHT:
+					right = true;
+					break;
+				case CELL_PAD_CTRL_DOWN:
+					down = true;
+					break;
+				case CELL_PAD_CTRL_LEFT:
+					left = true;
+					break;
+				default:
+					break;
+				}
+			}
+			else if (button.m_offset == CELL_PAD_BTN_OFFSET_DIGITAL2)
+			{
+				switch (button.m_outKeyCode)
+				{
+				case CELL_PAD_CTRL_SQUARE:
+					buf[0] |= (1 << 0);
+					break;
+				case CELL_PAD_CTRL_CROSS:
+					buf[0] |= (1 << 1);
+					break;
+				case CELL_PAD_CTRL_CIRCLE:
+					buf[0] |= (1 << 2);
+					break;
+				case CELL_PAD_CTRL_TRIANGLE:
+					buf[0] |= (1 << 3);
+					break;
+				case CELL_PAD_CTRL_PS:
+					buf[1] |= (1 << 4);
+					break;
+				default:
+					break;
+				}
+			}
+		}
+	}
+
+	if (!up && !right && !down && !left)
+		buf[0x02] = 0x0f;
+	else if (up && !left && !right)
+		buf[0x02] = 0x00;
+	else if (up && right)
+		buf[0x02] = 0x01;
+	else if (right && !up && !down)
+		buf[0x02] = 0x02;
+	else if (down && right)
+		buf[0x02] = 0x03;
+	else if (down && !left && !right)
+		buf[0x02] = 0x04;
+	else if (down && left)
+		buf[0x02] = 0x05;
+	else if (left && !up && !down)
+		buf[0x02] = 0x06;
+	else if (up && left)
+		buf[0x02] = 0x07;
+
+
+	auto& mouse_handler = g_fxo->get<MouseHandlerBase>();
+	std::lock_guard mouse_lock(mouse_handler.mutex);
+
+	mouse_handler.Init(1);
+
+	const MouseInfo& current_info = mouse_handler.GetInfo();
+	MouseDataList& data_list = mouse_handler.GetDataList(0);
```

Comment:
This makes no sense.
Just use the mouse and its members
```suggestion
	const Mouse& mouse = ::at32(handler.GetMice(), 0);
```

Reply:
done